tsl/src/remote/dist_ddl.c

```diff
@@ -471,19 +471,20 @@ dist_ddl_preprocess(ProcessUtilityArgs *args)
 		case T_TruncateStmt:
 		{
 			TruncateStmt *stmt = (TruncateStmt *) args->parsetree;
-			/* Calculate number of regular tables. Note that distributed
-			 * hypertables are filtered from the relations list in
-			 * process_utility.c, so the list might actually be empty. */
+
+			/* Calculate number of regular tables. Note that number of
+			 * distributed hypertables should be the same of the relations
+			 * list */
 			unsigned int num_regular_tables =
 				list_length(stmt->relations) - num_hypertables - num_dist_hypertable_members;
 
-			Assert(num_regular_tables <= list_length(stmt->relations));
+			Assert(num_regular_tables == list_length(stmt->relations));
```

Comment:
Why are these going to be equal? If there are hypertables in the list, I would assume that the number of regular tables can be less than the total number of relations.

Reply:
Ooops... forgot to revert it during coding!